transaction/std/src/test_utils.rs

```diff
@@ -201,3 +206,72 @@ pub fn get_transaction<RNG: RngCore + CryptoRng, FPR: FogPubkeyResolver + Clone>
 
     transaction_builder.build(rng)
 }
+
+/// Build simulated change memo with zero amount
+pub fn build_zero_value_change_memo(
+    builder: &mut impl MemoBuilder,
+) -> Result<MemoPayload, NewMemoError> {
+    build_change_memo_with_amount(builder, Amount::new(0, 0.into()))
+}
+
+/// Build simulated change memo with amount
+pub fn build_change_memo_with_amount(
+    builder: &mut impl MemoBuilder,
+    change_amount: Amount,
+) -> Result<MemoPayload, NewMemoError> {
+    // Create simulated context
+    let mut rng: StdRng = SeedableRng::from_seed([0u8; 32]);
+    let alice = AccountKey::random_with_fog(&mut rng);
+    let alice_address_book = ReservedDestination::from(&alice);
+    let change_tx_pubkey = RistrettoPublic::from_random(&mut rng);
+    let memo_context = MemoContext {
+        tx_public_key: &change_tx_pubkey,
+    };
+
+    //Build memo
+    builder.make_memo_for_change_output(change_amount, &alice_address_book, memo_context)
+}
+
+/// Utilities for decoding and verifying memo data
+pub struct MemoDecoder;
+
+impl MemoDecoder {
+    /// Decode a sender note from bytes
+    pub fn decode_sender_note(memo_data: &[u8; GiftCodeSenderMemo::MEMO_DATA_LEN]) -> &str {
```

Comment:
hmm so, why do we need this, can we just do `GiftCodeSenderMemo::try_from(MemoPayload)` ?